### PR TITLE
Refactor WIP

### PR DESCRIPTION
**Pull request by alexdev03**
- [x] Remove Annotaml in favor of ConfigLib
- [x] Add lombok and begin using this everywhere
- [x] Separate Settings into tab_groups.yml and config.yml
- [x] tab_groups.yml - should be a TabGroups class in the config package containing a List of Group objects - @Configuration annotated
- [x] config.yml - should be a Settings class containing all the other plugin options
- [x] Refactor logic to use interface Providers (see HuskClaims)
- [x] e.g. a ConfigProvider providing get/setSettings(), get/setTabGroups(), a ScoreboardProvider for sorting, manager providers for other parts, and so on...
- [x] Rework manager classes to accept Group objects instead of individually getting stuff from Maps in settings

### CODE REVIEW

File: src/main/java/net/william278/velocitab/packet/PlayerChannelHandler.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            super.write(ctx, msg, promise);
+            return;
+        }
+
+        if(!minecraftPacket.containsAction(UpsertPlayerInfo.Action.ADD_PLAYER) && !minecraftPacket.containsAction(UpsertPlayerInfo.Action.UPDATE_LISTED)) {
+            super.write(ctx, msg, promise);
+            return;
+        }
+
+        if (minecraftPacket.getEntries().stream().allMatch(entry -> entry.getProfile() != null && entry.getProfile().getName().startsWith("CIT"))) {
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/packet/PlayerChannelHandler.java:48]: perhaps make CIT a const here?

File: src/main/java/net/william278/velocitab/packet/ChannelManager.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import net.william278.velocitab.player.TabPlayer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class ChannelManager {
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/packet/ChannelManager.java:44]: hmm, `PacketEventManager`? might be more appropriate?

File: src/main/java/net/william278/velocitab/player/TabPlayer.java [hunk 3]
```diff
@@ -172,4 +172,18 @@ public boolean equals(Object obj) {
...
+                ", role=" + role +
+                ", headerIndex=" + headerIndex +
+                ", footerIndex=" + footerIndex +
+                ", lastDisplayname=" + lastDisplayname +
+                ", teamName='" + teamName + '\'' +
+                ", lastServer='" + lastServer + '\'' +
+                ", group=" + group.name() +
+                ", loaded=" + loaded +
+                '}';
+    }
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/player/TabPlayer.java:188]: What are we using this for? Debug?

Comment (alexdev03) [reply src/main/java/net/william278/velocitab/player/TabPlayer.java:188]: Yes

File: src/main/java/net/william278/velocitab/packet/PlayerChannelHandler.java (hunk 1 above)
Comment (alexdev03) [reply src/main/java/net/william278/velocitab/packet/PlayerChannelHandler.java:48]: CIT is for citizens that handles npcs in a bad way.

**Review by WiIIiam278 — APPROVED** [review]
Excellent work :)

**Review by WiIIiam278 — APPROVED** [review]

**Pull request merged**